File: backend/src/auth/domain/core/auth_repository.ts
import { Either } from "funpjs";
import AuthFailure from "./auth_failure";
import AuthResult from "./auth_result";

export default interface AuthRepository {
  signUpWithEmailAndPassword(
    email: string,
    password: string
  ): Promise<Either<AuthFailure, AuthResult>>;
  signInWithEmailAndPassword(
    email: string,
    password: string
  ): Promise<Either<AuthFailure, AuthResult>>;
  refreshAccessToken(
    refreshToken: string
  ): Promise<Either<AuthFailure, string>>;
  // signUpWithGoogle(): Promise<Either<AuthFailure, UserEntity>>;
}

File: app/lib/domain/auth/auth_repository.dart
import 'package:dartz/dartz.dart';
import 'package:telleo/domain/user/user_entity.dart';

import 'auth_failure.dart';

abstract class AuthRepository {
  Future<Either<AuthFailure, UserEntity>> signUpWithEmailAndPassword({
    required EmailAdress email,
    required Password password,
  });
  Future<Either<AuthFailure, UserEntity>> signInWithEmailAndPassword({
    required EmailAdress email,
    required Password password,
  });

  Future<Either<AuthFailure, UserEntity>> signInWithGoogle();
}
